acl/src/main/java/org/apache/rocketmq/acl/plain/RemoteAddressStrategyFactory.java

```diff
@@ -160,21 +221,33 @@ private void setValue(int start, int end) {
         @Override
         public boolean match(PlainAccessResource plainAccessResource) {
             String netAddress = plainAccessResource.getWhiteRemoteAddress();
-            if (netAddress.startsWith(this.head)) {
-                String value;
-                if (index == 3) {
-                    value = netAddress.substring(this.head.length());
-                } else {
-                    value = netAddress.substring(this.head.length(), netAddress.lastIndexOf('.'));
+            InetAddressValidator validator = InetAddressValidator.getInstance();
+            if (validator.isValidInet4Address(netAddress)) {
+                if (netAddress.startsWith(this.head)) {
+                    String value;
+                    if (index == 3) {
+                        value = netAddress.substring(this.head.length());
+                    } else if (index == 2) {
+                        value = netAddress.substring(this.head.length(), netAddress.lastIndexOf('.'));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the lastIndexOf() search on line 231 does not find the substring and returns -1, the call to substring() on line 231 will throw IndexOutOfBoundsException.